lib/dns_wrapper.rb

```diff
@@ -12,46 +12,20 @@ def dns_client
   
   public
   
-  def lookup_dns(zone_id, domain_name)
-    aws_ip = lookup_dns_a_record(zone_id, domain_name)
-    dns_ip = Resolv.getaddress(domain_name)
-    fail("Discrepancy for #{domain_name}: AWS=#{aws_ip} but DNS=#{dns_ip}") unless dns_ip == aws_ip
-    dns_ip
-  end
-  
-  def update_dns_a_record(zone_id, domain_name, new_ip)
-    update_response = request_update_dns_a_record(zone_id, domain_name, new_ip)
-    
-    1.upto(WAIT_ATTEMPTS) do |try|
-      break if update_insync?(update_response)
-      fail('Giving up') if try >= WAIT_ATTEMPTS
-      LOGGER.info("try #{try}: DNS update not yet propagated to AWS nameservers...")
-      sleep(WAIT_INTERVAL)
-    end
-    
-    1.upto(WAIT_ATTEMPTS) do |try|
-      break if Resolv.getaddress(domain_name) == new_ip
-      fail('Giving up') if try >= WAIT_ATTEMPTS
-      LOGGER.info("try #{try}: DNS update not yet propagated to local nameserver...")
-      sleep(WAIT_INTERVAL)
+  def lookup_cname(zone_id, name)
+    cname_from_dns = Resolv::DNS.new.getresource(name, Resolv::DNS::Resource::IN::CNAME).name.to_s
+    cname_from_aws = lookup_dns_cname_record(zone_id, name)
+    if cname_from_dns != cname_from_aws
+      fail("CNAME from DNS (#{cname_from_dns}) != CNAME from AWS (#{cname_from_aws})")
     end
+    cname_from_aws
   end
```

Comment:
- Go to [ELB in AWS](https://console.aws.amazon.com/ec2/v2/home?region=us-east-1#LoadBalancers:)
- Click on an entry
- Look at the first line of the details:
> DNS Name: 	
T2-11-12-wgbh-mla-test-org-A-258535996.us-east-1.elb.amazonaws.com (A Record)

So it's a bad choice of names on my part: `cname_target_from_aws` would be clear about the intent, but just plain `dns_name` would be better. hold on...